### PR TITLE
Added Disjoint-Set-Union.go to the datastructures folder

### LINKED ISSUE #185: Add A DSU data Structure Implementation in GoLang
I have implemented Disjoint Set Union in GoLang.
Wanted to Contribute here.

### PR DESCRIPTION
**Pull request by kalashshah11**
Closes #185  Implemented Disjoint Set Union in GoLang

### CODE REVIEW

File: data-structures/Disjoint-Set-Union/Disjoint-Set-Union.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	} else if d.rank[xroot] > d.rank[yroot] {
+		d.parent[yroot] = xroot
+	} else {
+		d.parent[yroot] = xroot
+		d.rank[xroot]++
+	}
+	return true
+}
+
+func main() {
```
Comment (nomisrevol) [inline data-structures/Disjoint-Set-Union/Disjoint-Set-Union.go:99]: Remove `main` function and create a test file instead.

**Review by nomisrevol — CHANGES_REQUESTED** [review]

File: data-structures/Disjoint-Set-Union/Disjoint-Set-Union.go (hunk 1 above)
Comment (kalashshah11) [reply data-structures/Disjoint-Set-Union/Disjoint-Set-Union.go:99]: What am I to add in the test File ?

Comment (kalashshah11) [reply data-structures/Disjoint-Set-Union/Disjoint-Set-Union.go]: Tried Adding The Tests Please Review.

**Comment by nomisrevol** [comment]
Please see other source files in `sorts` to see how to name the implementation file and test file. We should keep the naming convention consistently. Please see how to write a test file (unit test) in Golang also. Prefer using lowercase instead of uppercase for naming anything, after that, please ask me to review again, I will review the implementation soon. Thank you.

**Comment by vedantmamgain** [comment]
Please ensure that the structure of the packages is apt. Your pr is currently failing the linter.

**Pull request closed without merging**